### PR TITLE
[Bifrost] local loglet append benchmark

### PR DESCRIPTION
**Pull request by AhmedSoliman**
[Bifrost] local loglet append benchmark

Adds a benchmark for the append throughput of the local loglet. The benchmark measures default config performance (throughtput and latency) for:
- Sequential single log appends (one at a time)
- Concurrent single log appends
- Concurrent multi-log appends

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/restatedev/restate/pull/1401).
* #1402
* __->__ #1401

### CODE REVIEW

File: crates/bifrost/benches/append_throughput.rs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+use restate_types::config::{
+    BifrostOptionsBuilder, CommonOptionsBuilder, ConfigurationBuilder, LocalLogletOptionsBuilder,
+};
+use restate_types::logs::{LogId, Payload};
+use tracing::info;
+use tracing_subscriber::EnvFilter;
+mod util;
+
+async fn append_records_multi_log(bifrost: Bifrost, log_id_range: Range<u64>, count_per_log: u64) {
+    // for every log there is N streams. total count_per_log / streams_per_log gives the number of
```
Comment (tillrohrmann) [inline crates/bifrost/benches/append_throughput.rs:28]: What is `streams_per_log` referring to?

File: crates/bifrost/benches/append_throughput.rs [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+    tracing_subscriber::fmt()
+        .with_env_filter(EnvFilter::from_default_env())
+        .init();
+
+    let tmp_base = tempfile::tempdir().unwrap();
+
+    let provider = restate_types::logs::metadata::ProviderKind::Local;
+    let total_num_logs = 64;
+    let num_logs_per_run = 10;
```
Comment (tillrohrmann) [inline crates/bifrost/benches/append_throughput.rs:79]: Possible follow-up: What I found helpful when playing around with different configurations is to make these values configurable via env variables or so. That way one does not have to rebuild the benchmark every time one wants to test a different configuration.

File: crates/bifrost/benches/append_throughput.rs [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+                .expect("config"),
+        )
+        .bifrost(
+            BifrostOptionsBuilder::default()
+                .local(local_opts)
+                .build()
+                .unwrap(),
+        )
+        .build()
+        .expect("config");
```
Comment (tillrohrmann) [inline crates/bifrost/benches/append_throughput.rs:96]: Possible follow-up: Loading other environment variable configured config options I found to be really helpful when trying to test different configurations.

File: crates/bifrost/benches/append_throughput.rs (hunk 1 above)
Comment (AhmedSoliman) [reply crates/bifrost/benches/append_throughput.rs:28]: An old iteration of this test, I'll remove.

File: crates/bifrost/benches/append_throughput.rs (hunk 2 above)
Comment (AhmedSoliman) [reply crates/bifrost/benches/append_throughput.rs:79]: Absolutely, good idea, might do that on a follow up.

**Pull request merged**